### ISSUE TITLE
"No such file or directory" error

### ISSUE THREAD

**Issue by justindsmith** [dherges/ng-packagr#166]
## Type of Issue

- [X] Bug Report
- [ ] Feature Request

## Description

This is directly related to issue #81.

While using the https://github.com/vmware/clarity/ components the `ng-packagr` step results in the following error output:

```
$ yarn build:lib
yarn build:lib v1.0.2
$ ng-packagr
Building Angular library from /Users/jsmith/Projects/ng-packaged/ng-package.json

BUILD ERROR
ENOENT: no such file or directory, open '/Users/jsmith/Projects/ng-packaged/node_modules/clarity-angular/icon/index.ts'
Error: ENOENT: no such file or directory, open '/Users/jsmith/Projects/ng-packaged/node_modules/clarity-angular/icon/index.ts'

error Command failed with exit code 111.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

I'm not sure if this is an issue in `ng-packagr` or in how `clarity` is packaged.

#### How To Reproduce

I've forked the `ng-packaged` repo to provide a reproducible environment: https://github.com/justindsmith/ng-packaged-clarity

You can see the commit I added for the only changes: https://github.com/justindsmith/ng-packaged-clarity/commit/238d76b4e968ca700f31edae67cea8cf4554747f

#### Version Information

```
ng-packagr: v1.3.0
node: v8.6.0
@angular: v4.4.4
rxjs: v5.4.2
zone.js: v0.8.17
```

**Comment by dherges** [comment]
Hi, can you check #129 and specially the `lib.externals` whether the issue is answered?

**Comment by justindsmith** [comment]
That gets me closer and meets my immediate needs (which is awesome!), but doesn't answer the question of getting all of Clarity added to a lib module.

The way clarity is organized there is a root module (`ClarityModule`) and then a bunch of sub-modules underneath that. I was not able to get the root module to work, even after providing the externals for the import as below (found in the clarity umd file).

```
"externals": {
  "clarity-angular": "ng.clarity"
}
```
I still get the same error.

If I ignore the root module and pull in only the sub-modules that I need (in this case `ClrIconModule`) then I can get it to work by defining the following externals:

```
"externals": {
  "clarity-angular/icon/icon.module": "ng.clarity"
}
```

My guess is that it's a problem in how clarity is joining the dependencies between the different sub-modules?

The packaging is catching on the `clarity-angular/icon/icon` export/import, which is the `IconCustomTag`. Looking at the clarity umd file, it appears that the `IconCustomTag` is not exported except inside an array `ICON_DIRECTIVE`, so I'm not immediately sure how to get at that dependency through the externals...

So for my _immediate_ need I was able to use the externals to get to the module that I cared about, which means "bug fixed", but I'm still not able to use the ClarityModule root module, which is weird (but might be an issue in how Clarity is packaging their modules).
